ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/events/CreateTableEvent.java

```diff
@@ -78,6 +87,12 @@ private List<HivePrivilegeObject> getOutputHObjs() {
     Table                     table = event.getTable();
     Database                  database = event.getDatabase();
     String                    uri   = getSdLocation(table.getSd());
+    // if the table is storagehandler based table, need storageuri as hive privilege objects
+    //  storagehandler type + cluster + location
+    String storagehandler_uri = table.getParameters().getOrDefault("storage_handler", null);
+    if (storagehandler_uri != null && storagehandler_uri.contains("iceberg")){
+      storagehandler_uri = "hiveicebergstoragehandler://"+ table.getParameters().getOrDefault("metadata_location", "");
+    }
```

Comment:
Implement the above suggestion here and remove the above HivePrivilegeObject in getInputHObjs()